primitives/src/merkle_tree/field_based_mht/path.rs

```diff
@@ -375,6 +415,104 @@ impl<T: FieldBasedMerkleTreeParameters> PartialEq<FieldBasedMHTPath<T>>
     for FieldBasedBinaryMHTPath<T>
 {
     fn eq(&self, other: &FieldBasedMHTPath<T>) -> bool {
-        self == other
+
+        // Paths must have the same length
+        if self.get_length() != other.get_length() {
+            return false;
+        }
+
+        // Iterate through both paths (NOTE: Could work also converting one of the two paths into
+        // the other type and then calling the corresponding Eq, but you save one iteration in this
+        // way)
```

Comment:
I would improve the clarity of the comment by saying that in this way you avoid sweeping twice the paths: saving one iteration seems a bit too generic

Reply:
Done